File: iot/simulator.py
import random
import time
import paho.mqtt.client as mqtt

BROKER = "localhost"
PORT = 1883

bins = [1, 2, 3]

client = mqtt.Client()
client.connect(BROKER, PORT, 60)

print("Simulatore avviato...")

while True:
    for bin_id in bins:
        fill = random.randint(0, 100)

        topic = f"bins/{bin_id}/fill_level"
        client.publish(topic, str(fill))

        print(f"Inviato: {topic} -> {fill}%")

    time.sleep(5)

File: iot/simulator.js
const mqtt = require('mqtt');

// Connessione al broker
const client = mqtt.connect('mqtt://localhost:1883');

// Lista bidoni
const bins = [1, 2, 3];

client.on('connect', () => {
  console.log('Simulatore connesso al broker MQTT');

  // Ogni 5 secondi invia dati
  setInterval(() => {
    bins.forEach((id) => {
      const fill = Math.floor(Math.random() * 100);

      const topic = `bins/${id}/fill_level`;

      client.publish(topic, fill.toString());

      console.log(`Inviato: ${topic} -> ${fill}%`);
    });
  }, 5000);
});
